### PR TITLE
BaSP-TG-32: Test Timesheets: get, getById, post

### PR DESCRIPTION
**Pull request by joryapur**

### CODE REVIEW

File: src/test/timesheets.test.js [hunk 1]
```diff
@@ -1 +1,117 @@
...
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Found timesheet with id 63531fd7410c845909ab22e7');
+  });
+
+  test('Should return 404 status when there is no time-sheet with a valid id ', async () => {
+    const response = await request(app).get(`/time-sheets/${nonExistentId}`).send();
+
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBe(true);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:37]: We should also test the message of the body

File: src/test/timesheets.test.js [hunk 2]
```diff
@@ -1 +1,117 @@
...
+  });
+
+  test('Should return 400 status when a invalid id is received ', async () => {
+    const response = await request(app).get(`/time-sheets/${invalidId}`).send();
+
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe(true);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:44]: We should also test the message of the body

File: src/test/timesheets.test.js [hunk 3]
```diff
@@ -1 +1,117 @@
...
+});
+
+describe('POST /time-sheets', () => {
+  test('Should create a time-sheet when all valid data is entered', async () => {
+    const response = await request(app).post('/time-sheets').send(mockedTimesheets);
+
+    // eslint-disable-next-line no-underscore-dangle
+    idTimesheet = response.body.data._id;
+
+    expect(response.status).toBe(201);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:55]: We should also test the message and error of the body

File: src/test/timesheets.test.js [hunk 4]
```diff
@@ -1 +1,117 @@
...
+    const response = await request(app).post('/time-sheets').send();
+
+    expect(response.status).toBe(400);
+  });
+
+  test('Should not create a time-sheet if date is empty', async () => {
+    mockedTimesheets.date = '';
+    const response = await request(app).post('/time-sheets').send(mockedTimesheets);
+
+    expect(response.status).toBe(400);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:68]: We should also test the message and error of the body

File: src/test/timesheets.test.js [hunk 5]
```diff
@@ -1 +1,117 @@
...
+    mockedTimesheets.date = '2022-10-22T00:00:00.000+00:00';
+  });
+
+  test('Should not create a time-sheet if task is empty', async () => {
+    mockedTimesheets.task = '';
+    const response = await request(app).post('/time-sheets').send(mockedTimesheets);
+
+    expect(response.status).toBe(400);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:76]: We should also test the message and error of the body

File: src/test/timesheets.test.js [hunk 6]
```diff
@@ -1 +1,117 @@
...
+    mockedTimesheets.task = '63531a7c73636855c2aa7f9a';
+  });
+
+  test('Should not create a time-sheet if project is empty', async () => {
+    mockedTimesheets.project = '';
+    const response = await request(app).post('/time-sheets').send(mockedTimesheets);
+
+    expect(response.status).toBe(400);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:84]: We should also test the message and error of the body

File: src/test/timesheets.test.js [hunk 7]
```diff
@@ -1 +1,117 @@
...
+    mockedTimesheets.project = '63531a7c73636855c2aa7f9a';
+  });
+
+  test('Should not create a time-sheet if employee is empty', async () => {
+    mockedTimesheets.employee = '';
+    const response = await request(app).post('/time-sheets').send(mockedTimesheets);
+
+    expect(response.status).toBe(400);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:92]: We should also test the message and error of the body

File: src/test/timesheets.test.js [hunk 8]
```diff
@@ -1 +1,117 @@
...
+
+  test('Should return 404 status when there are no thime-sheets ', async () => {
+    await request(app).delete(`/time-sheets/${timeSheetIdSeed}`).send();
+    await request(app).delete(`/time-sheets/${idTimesheet}`).send();
+
+    const response = await request(app).get('/time-sheets').send();
+
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBe(true);
+    expect(response.body.message).toBe('Timesheets not found');
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:115]: We could also test that the data of the body is empty

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by nicolasblasco — APPROVED** [review]
